locksmith/src/payment/paymentProcessor.ts

```diff
@@ -51,27 +51,26 @@ export class PaymentProcessor {
       const user = await this.findUserByPublicKey(publicKey)
       const stripeCustomerId = await getStripeCustomerIdForAddress(publicKey)
 
-      if (user && stripeCustomerId) {
+      // If we already have a stripe customer id
+      if (stripeCustomerId) {
         await this.stripe.customers.createSource(stripeCustomerId, {
           source: token,
         })
 
         return true
       }
-      if (user && !stripeCustomerId) {
-        const customer = await this.createStripeCustomer(
-          user.emailAddress,
-          token
-        )
-        return !!(await saveStripeCustomerIdForAddress(publicKey, customer.id))
-      }
-      return false
+
+      const customer = await this.createStripeCustomer(
+        token,
+        user ? user.emailAddress : ''
```

Comment:
In the case of no user available is it okay to pass empty string here?

Reply:
yes